RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFClusterSoAProducer.cc

```diff
@@ -0,0 +1,69 @@
+#include <Eigen/Core>
+#include <Eigen/Dense>
+#include "DataFormats/ParticleFlowReco/interface/PFRecHitHostCollection.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/stream/EDProducer.h"
+#include "HeterogeneousCore/CUDACore/interface/JobConfigurationGPURecord.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/alpaka/PFClusterParamsDeviceCollection.h"
+#include "RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFClusterSoAProducerKernel.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/PFCPositionCalculatorBase.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  class PFClusterSoAProducer : public stream::EDProducer<> {
+  public:
+    PFClusterSoAProducer(edm::ParameterSet const& config)
+        : pfClusParamsToken(esConsumes(config.getParameter<edm::ESInputTag>("pfClusterParams"))),
+          inputPFRecHitSoA_Token_{consumes(config.getParameter<edm::InputTag>("pfRecHits"))},
+          outputPFClusterSoA_Token_{produces()},
+          outputPFRHFractionSoA_Token_{produces()},
+          synchronise_(config.getParameter<bool>("synchronise")),
+          pfRecHitFractionAllocation_(config.getParameter<int>("pfRecHitFractionAllocation")) {}
+
+    void produce(device::Event& event, device::EventSetup const& setup) override {
+      const reco::PFClusterParamsDeviceCollection& params = setup.getData(pfClusParamsToken);
+      const reco::PFRecHitHostCollection& pfRecHits = event.get(inputPFRecHitSoA_Token_);
+      const int nRH = pfRecHits->size();
+
+      reco::PFClusteringVarsDeviceCollection pfClusteringVars{nRH, event.queue()};
+      reco::PFClusteringEdgeVarsDeviceCollection pfClusteringEdgeVars{(nRH * 8), event.queue()};
```

Comment:
these had a `+ 1` to the size, is it not needed any more ?

Reply:
No, not needed any more. It was a holdover from some debugging a long time ago.